controllers/registry_controller.go

```diff
@@ -94,7 +95,20 @@ func (r *registryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("unable to create signer from private key secret: %w", err)
 	}
-	for _, node := range nodes.Items {
+	for _, n := range nodes.Items {
+		// Ensure the node status is up to date, going through this list could take a long time depending on the
+		// number of Windows nodes in the cluster
+		var node core.Node
+		if err := r.client.Get(ctx, types.NamespacedName{Name: n.GetName()}, &node); err != nil {
+			if k8sapierrors.IsNotFound(err) {
+				continue
```

Comment:
I feel that looking after the present of the node object is good enough. As a plus we can track the status of  [Unschedulable](https://github.com/openshift/windows-machine-config-operator/blob/3d46617e9c4295ca5ff551ea7a4c35707a5667bb/vendor/k8s.io/api/core/v1/types.go#L6032) if we need a more granular approach with Manual Node administration

Reply:
https://kubernetes.io/docs/concepts/architecture/nodes/#manual-node-administration

Reply:
Agree